src/tests/fs_tests.rs

```diff
@@ -1083,7 +1243,13 @@ pub mod fs_tests {
         lindrustfinalize();
     }
 
+    #[test]
     pub fn ut_lind_fs_shm() {
+        //acquiring a lock on TESTMUTEX prevents other tests from running concurrently
+        let mut _thelock = TESTMUTEX.lock().unwrap();
+        //creates the file system if it doesnt exist yet.
+        setup::test_setup();
```

Comment:
Would it be better to put this in a single "lock_and_init" function?  That way if you want to add something else, you don't need to go and modify the code everywhere?

(I would think that `lindrustinit(0);` would also be part of this...)

Reply:
we could try something like that, but then it will be like a semaphore ig rather than a lock. since lock is tied to scope. that wouldn't work as it will go out of scope after lock_and_init call?

Reply:
Okay, I don't know enough about this, but that makes sense.  

What about combining setup::test_setup(); and lindrustinit(0); into a helper?

Reply:
will explore the helper function idea!

Reply:
implemented a lock_and_init function as suggested.